MSAL/src/MSALPublicClientApplication.m

```diff
@@ -99,13 +113,17 @@ - (id)initWithClientId:(NSString *)clientId
     CHECK_RETURN_NIL([self generateRedirectUriWithClientId:_clientId
                                                      error:error]);
     
-    id<MSALTokenCacheAccessor> dataSource;
+    id<MSIDTokenCacheDataSource> dataSource;
 #if TARGET_OS_IPHONE
-    dataSource = [MSALKeychainTokenCache defaultKeychainCache];
+    dataSource = MSIDKeychainTokenCache.defaultKeychainCache;
 #else
-    dataSource = [MSALWrapperTokenCache defaultCache];
+    dataSource = MSIDMacTokenCache.defaultCache;
 #endif
-    _tokenCache = [[MSALTokenCache alloc] initWithDataSource:dataSource];
+    
+    MSIDLegacyTokenCacheAccessor *legacyAccessor = [[MSIDLegacyTokenCacheAccessor alloc] initWithDataSource:dataSource];
+    MSIDDefaultTokenCacheAccessor *defaultAccessor = [[MSIDDefaultTokenCacheAccessor alloc] initWithDataSource:dataSource];
+    
+    self.tokenCache = [[MSIDSharedTokenCache alloc] initWithPrimaryCacheAccessor:defaultAccessor otherCacheAccessors:@[legacyAccessor]];
```

Comment:
both accessors should be only present for iOS for now

Reply:
done